src/components/CPU/BreakpointPanel.js

```diff
@@ -0,0 +1,209 @@
+			'connection': () => fetchBreakpoints(),
+			'cpu.stepping': () => fetchBreakpoints(),
+			'cpu.breakpoint.add': () => fetchCpuBreakpoints(),
+			'cpu.breakpoint.update': () => fetchCpuBreakpoints(),
+			'cpu.breakpoint.remove': () => fetchCpuBreakpoints(),
+			'memory.breakpoint.add': () => fetchMemoryBreakpoints(),
+			'memory.breakpoint.update': () => fetchMemoryBreakpoints(),
+			'memory.breakpoint.remove': () => fetchMemoryBreakpoints(),
+		});
+		return () => listeners.forget(listeners_);
+	}, [fetchBreakpoints, fetchCpuBreakpoints, fetchMemoryBreakpoints, clearBreakpoints]);
+
+	const handleToggleCpuBreakpoint = (breakpoint) => {
+		context.ppsspp.send({
+			event: 'cpu.breakpoint.update',
+			address: breakpoint.address,
+			enabled: !breakpoint.enabled,
+		});
+	};
+
+	const handleToggleMemoryBreakpoint = (breakpoint) => {
+		context.ppsspp.send({
+			event: 'memory.breakpoint.update',
+			address: breakpoint.address,
+			size: breakpoint.size,
+			enabled: !breakpoint.enabled,
+		});
+	};
+
+	const handleToggleBreakpoint = (breakpoint, type) => {
+		if (type === 'memory') {
+			handleToggleMemoryBreakpoint(breakpoint);
+		} else if (type === 'execute') {
+			handleToggleCpuBreakpoint(breakpoint);
+		}
+	};
+
+	const handleCloseBreakpointModal = () => {
+		setEditingBreakpoint(null);
+		setCreatingBreakpoint(false);
+	};
+
+	const onKeyDown = (ev) => {
+		if (hasContextMenu()) {
+			return;
+		}
+
+		if (ev.key === 'ArrowUp' && selectedRow > 0) {
+			setSelectedRow(selectedRow - 1);
+			ev.preventDefault();
+		}
+		if (ev.key === 'ArrowDown' && selectedRow < breakpoints.length - 1) {
+			setSelectedRow(selectedRow + 1);
+			ev.preventDefault();
+		}
+		const selectedBreakpoint = breakpoints[selectedRow];
+		if (ev.key === ' ' && selectedBreakpoint) {
+			handleToggleBreakpoint(selectedBreakpoint, selectedBreakpoint.size ? 'memory' : 'execute');
+			ev.preventDefault();
+		}
+		if (ev.key === 'Enter' && selectedBreakpoint) {
+			setEditingBreakpoint(selectedBreakpoint);
+			ev.preventDefault();
+		}
+		if (ev.key === 'Delete' && selectedBreakpoint) {
+			if (selectedBreakpoint.size) {
+				context.ppsspp.send({
+					event: 'memory.breakpoint.remove',
+					address: selectedBreakpoint.address,
+					size: selectedBreakpoint.size,
+				});
+			} else {
+				context.ppsspp.send({
+					event: 'cpu.breakpoint.remove',
+					address: selectedBreakpoint.address,
+				});
+			}
+			ev.preventDefault();
+		}
+	};
+
+	return (
+		<>
+			<ContextMenuTrigger id="breakpointPanel" renderTag="div" holdToDisplay={-1} attributes={{ id: 'BreakpointPanel' }}>
+				<table className="BreakpointPanel__table" onKeyDown={onKeyDown} tabIndex={0}>
+					<thead>
+						<tr>
+							<th>Break</th>
+							<th>Type</th>
+							<th>Offset</th>
+							<th>Size/Label</th>
+							<th>Opcode</th>
+							<th>Condition</th>
+							<th>Hits</th>
+						</tr>
+					</thead>
+					<colgroup>
+						<col style={{ width: '1%' }}/>

```

Comment:
This should be in CSS - it can either be in a colgroup or on the headers directly, using classes.  For example, see the VisualizeVFPU component.  It'd be fine to set all the 18% and override just __toggle and __hits or something.

-[Unknown]